### ISSUE TITLE
QR Codes should be supported in templates 

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/templating#44]
[Enabling QR Code generation for authenticator apps](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity-enable-qrcodes)

That should be supported in the templates like [SMS 2FA](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/2fa) used to be. 

requested by customer in [Enabling QR Code generation for authenticator apps](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity-enable-qrcodes) comments

**Comment by DamianEdwards** [comment]
This requires additional APIs in the platform (.NET Core) or external libraries. I don't wish to include the JS library in the templates just for the QR code. Similarly I don't want to include a 3rd-party server-side drawing library just to enable it either (e.g. `ImageSharp.Drawing`).

`System.Drawing` is being made available for .NET Core for compatibility purposes so we could investigate using it for this purpose.

**Comment by Eilon** [comment]
cc @blowdart

**Comment by blowdart** [comment]
System.Drawing is not suitable, as it's a straight port of the "do not use in a web app code"

**Comment by DamianEdwards** [comment]
Yeah, but people do anyway. Always have, always will. There isn't any plan for a System.* replacement AFAIK, so if we don't use it, we need to make something custom (not out of the question given it's just a QR code) or look at `ImageSharp.Drawing`.

**Comment by DamianEdwards** [comment]
ping @muratg who was looking at this too

**Comment by blowdart** [comment]
But _we_ shouldn't.

Anyway it's not cross platform.

**Comment by DamianEdwards** [comment]
It is? I couldn't remember if they limited that one to Windows or not. Many packages in the "Windows compat pack" actually have *nix implementations included as well.

**Comment by blowdart** [comment]
I think so, either way, we don't want to encourage the use of something that may leak memory and handles.

**Comment by DamianEdwards** [comment]
@blowdart so serious 😉 

Well then, given `ImageSharp.Drawing` is still only in beta, we're a bit stuck unless we want to go the route of doing a custom QR code generator.

**Comment by blowdart** [comment]
We are, so we stick with the current situation IMO

**Comment by Eilon** [comment]
Moving this to 2.2.0.

**Comment by mkArtakMSFT** [comment]
@blowdart, moving this out to 3.0 as I don't think this is a priority for 2.2
